src/libraries/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableArray_1.cs

```diff
@@ -791,6 +764,201 @@ public IEnumerable<TResult> OfType<TResult>()
             return self.array.OfType<TResult>();
         }
 
+        /// <summary>
+        /// Adds the specified values to this list.
+        /// </summary>
+        /// <param name="items">The values to add.</param>
+        /// <returns>A new list with the elements added.</returns>
+        public ImmutableArray<T> AddRange(ReadOnlySpan<T> items)
+        {
+            var self = this;
+            return self.InsertRange(self.Length, items);
+        }
+
+        /// <summary>
+        /// Adds the specified values to this list.
+        /// </summary>
+        /// <param name="items">The values to add.</param>
+        /// <returns>A new list with the elements added.</returns>
+        public ImmutableArray<T> AddRange(params T[] items)
+        {
+            var self = this;
+            return self.InsertRange(self.Length, items);
+        }
+
+        /// <summary>
+        /// Creates a <see cref="ReadOnlySpan{T}"/> over the portion of current <see cref="ImmutableArray{T}"/> beginning at a specified position for a specified length.
+        /// </summary>
+        /// <param name="start">The index at which to begin the span.</param>
+        /// <param name="length">The number of items in the span.</param>
+        /// <returns>The <see cref="ReadOnlySpan{T}"/> representation of the <see cref="ImmutableArray{T}"/></returns>
+        public ReadOnlySpan<T> AsSpan(int start, int length) => new ReadOnlySpan<T>(array, start, length);
+
+#if NETCOREAPP3_0_OR_GREATER || NETSTANDARD2_1_OR_GREATER
```

Comment:
Best practice suggests factoring the tfm-specific APIs to a separate file, e.g. `ImmutableArray_1.netcoreapp.cs`